File: hw5/examples/python/mysql_test_3.py

import mysql.connector as mc

import config

def main():
    try:
        # get connection info
        hst = config.host
        dat = config.database
        usr = config.user
        pwd = config.password
        
        # create connection
        cn = mc.connect(host=hst, database=dat, user=usr, password=pwd)
        
        # get a pet id, make sure it is unique
        pet_id = int(input('Please enter a pet id: '))
        q = "SELECT * FROM pet WHERE id = %s"
        rs = cn.cursor()
        rs.execute(q, (pet_id,))
        if rs.fetchone(): 
            print('This pet id is already taken')
            rs.close()
            cn.close()
            return

        # get new pet info
        name = input("Please enter the pet's name: ")
        pet_type = input("Please enter the pet's type (dog, cat, etc): ")
        breed = input("Please enter the pet's breed: ")
        bday = input("Please enter the pet's birthdate ('YYYY-MM-DD'): ")
        size = input("Please enter the pet's size (S, M, L, XL): ")
        appearance = input("Please enter the pet's appearance: ")

        # run update
        q = "INSERT INTO pet VALUES (%s,%s,%s,%s,%s,%s,%s)"
        rs = cn.cursor()
        rs.execute(q, (pet_id, name, pet_type, breed, bday, size, appearance))
        cn.commit()

    except mc.Error as err:
        print(err)

    finally:
        rs.close()
        cn.close()
            
        
if __name__ == '__main__':
    main()

File: hw5/examples/cpp/mysql_test_3.cpp

#include <iostream>
#include <mysql_connection.h>
#include <cppconn/driver.h>
#include <cppconn/prepared_statement.h>
#include <cppconn/resultset.h>
#include <cppconn/exception.h>

#include "config.h"

using namespace std;


int main()
{
  try {

    // connection info
    string hst = config::host;
    string usr = config::user;
    string pwd = config::password;
    string dab = config::database;

    // create connection
    sql::Driver* driver = get_driver_instance();
    sql::Connection* cn = driver->connect(hst, usr, pwd);
    cn->setSchema(dab);

    // get a pet id, make sure it is unique
    cout << "Please enter a pet id: ";
    int id;
    cin >> id;
    cin.ignore();
    string q = "SELECT * FROM pet WHERE id = ?";
    sql::PreparedStatement* st = cn->prepareStatement(q);
    st->setInt(1, id);
    sql::ResultSet* rs = st->executeQuery();
    if (rs->next()) {
      cout << "This pet id is already taken" << endl;
      delete rs;
      delete st;
      delete cn;
      return 1;
    }
    delete rs;
    delete st;
    
    // get pet name, breed, type, bday, and size
    cout << "Please enter the pet's name: ";
    string name;
    getline(cin, name);
    cout << "Please enter the pet's type (dog, cat, etc): ";
    string type;
    getline(cin, type);
    cout << "Please enter the pet's breed: ";
    string breed;
    getline(cin, breed);
    cout << "Please enter the pet's birthdate ('YYYY-MM-DD'): ";
    string date;
    getline(cin, date);
    cout << "Please enter the pet's size (S, M, L, XL): ";
    string size;
    getline(cin, size);
    cout << "Please enter the pet's appearance: ";
    string appearance;
    getline(cin, appearance);

    q = "INSERT INTO pet VALUES (?,?,?,?,?,?,?)";
    st = cn->prepareStatement(q);
    st->setInt(1, id);
    st->setString(2, name);
    st->setString(3, type);
    st->setString(4, breed);
    st->setString(5, date);
    st->setString(6, size);
    st->setString(7, appearance);
    st->executeUpdate();

    delete st;
    delete cn;
    
  } catch(sql::SQLException& e) {
    cout << e.what() << endl;
  }
  
}
